### PR TITLE
Extract common e2e functions and types

### LINKED ISSUE #265: [morpheus/tokenvm] move `e2e` helpers to a shared package
Blocked: #258

### PR DESCRIPTION
**Pull request by vikinatora**
# Context
The goal of this PR is to resolve #265. 

The approach I've taken is to extract common functions(`awaitHealthy`,`generateBlocks` and `acceptTransaction`) into a separate package I named `common`. This refactoring lead to changing the functions' signatures, as well as creating a few new interfaces and types. I hope that was the idea behind the original issue.

# Testing
I've ran both morpheusvm and tokenvm e2e tests by running `MODE=test ./scripts/run.sh` and observing that all tests pass.

### CODE REVIEW

File: common/types.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"context"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/rpc"
+)
+
+// RpcClient is an interface of common rpc client functions
+type RpcClient interface {
```
Comment (wlawt) [inline common/types.go:11]: nit: I would probably name it `RPCClient`

File: common/types.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	Parser(ctx context.Context) (chain.Parser, error)
+	WaitForTransaction(ctx context.Context, txID ids.ID) (bool, uint64, error)
+}
+
+// NodeInstance is a struct that holds information about a node
+type NodeInstance struct {
+	NodeID ids.NodeID
+	Uri    string
+	Cli    *rpc.JSONRPCClient
+	VmCli  interface{}
```
Comment (wlawt) [inline common/types.go:21]: nit: also would probably name it `VMCli`

File: common/functions.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
+package common
```
Comment (wlawt) [inline common/functions.go:1]: awesome! small nit, maybe `helpers.go` would be a clearer file name?

**Comment by vikinatora** [comment]
Thank you @wlawt for the naming suggestions. I've adhered to them since you're more familiar with the coding and naming style of hypersdk.

**Review by wlawt — COMMENTED** [review]
Did another pass, and everything seems fine to me. It would be good to get @patrick-ogrady 's thoughts on it though.

**Pull request closed without merging**